dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/BaseAnalyticalObject.java

```diff
@@ -480,14 +481,18 @@ public List<DataElement> getDataElements() {
             .collect(Collectors.toList()));
   }
 
-  /** Returns all indicators in the data dimensions. The returned list is immutable. */
+  /**
+   * Returns all indicators in the data dimensions. The returned set is immutable. Return as
+   * distinct set instead of list as there can be many data dimension items referencing the same
+   * indicator
+   */
   @JsonIgnore
-  public List<Indicator> getIndicators() {
-    return ImmutableList.copyOf(
+  public Set<Indicator> getIndicators() {
+    return ImmutableSet.copyOf(
```

Comment:
We can use `Set.copyOf` or collect to an unmodifiable set in the first place

Reply:
i'll update this